src/RestrictedPython/compile.py

```diff
@@ -107,32 +112,87 @@ def compile_restricted_single(
 
 
 def compile_restricted_function(
-        p,
+        p,  # parameters
         body,
         name,
         filename='<string>',
-        globalize=None,
+        globalize=None,  # List of globals (e.g. )
+        flags=0,
+        dont_inherit=False,
         policy=RestrictingNodeTransformer):
     """Compile a restricted code object for a function.
 
-    The function can be reconstituted using the 'new' module:
-
-    new.function(<code>, <globals>)
-
     The globalize argument, if specified, is a list of variable names to be
     treated as globals (code is generated as if each name in the list
     appeared in a global statement at the top of the function).
+    This allows to inject global variables into the generated function that
+    feel like they are local variables, so the programmer who uses this doesn't
+    have to understand that his code is executed inside a function scope
+    instead of the global scope of a module.
+
+    To actually get an executable function, you need to execute this code and
+    pull out the defined function out of the locals like this:
+
+    >>> compiled = compile_restricted_function('', 'pass', 'function_name')
+    >>> safe_locals = {}
+    >>> safe_globals = {}
+    >>> exec(compiled.code, safe_globals, safe_locals)
+    >>> compiled_function = safe_locals.values()[0]
+    >>> result = compiled_function(*[], **{})
+
+    Then if you want to controll the globals for a specific call to this
+    function, you can regenerate the function like this:
+
+    >>> my_call_specific_global_bindings = dict(foo='bar')
+    >>> safe_globals = safe_globals.copy()
+    >>> safe_globals.update(my_call_specific_global_bindings)
+    >>> import types
+    >>> new_function = types.FunctionType(compiled_function.func_code, \
+                                          safe_globals, \
+                                          '<function_name>', \
+                                          compiled_function.func_defaults or \
+                                          () \
+                                          )
+    >>> result = new_function(*[], **{})
```

Comment:
Is this test code actually run? Should it be moved to the Sphinx documentation where we could test it?